mesonbuild/mtest.py

```diff
@@ -1032,25 +1022,40 @@ def complete(self) -> None:
             self.stde += f'\n(test program exited with status code {self.returncode})'
         super().complete()
 
-    async def parse(self, harness: 'TestHarness', lines: T.AsyncIterator[str]) -> None:
+    async def parse(self,
+                    harness: 'TestHarness',
+                    lines: T.AsyncIterator[str],
+                    printer: T.Optional[T.Callable[[str], bool]] = None) -> None:
         res = None
 
         async for i in TAPParser().parse_async(lines):
             if isinstance(i, TAPParser.Bailout):
-                res = TestResult.ERROR
-                harness.log_subtest(self, i.message, res)
+                msg = harness.format_subtest(self, i.message, TestResult.ERROR, ConsoleLogger.SUB)
```

Comment:
The formatting code should not be part of TestRun subclasses. The printer callback should receive the parameters that are needed to call `harness.format_subtest`.